src/mscorlib/src/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -253,7 +253,12 @@ internal static CultureInfo GetNeutralResourcesLanguage(Assembly a, ref Ultimate
                 // fires, please fix the build process for the BCL directory.
                 if (a == typeof(Object).Assembly)
                 {
-                    Contract.Assert(false, "mscorlib's NeutralResourcesLanguageAttribute is a malformed culture name! name: \"" + cultureName + "\"  Exception: " + e);
+#if FEATURE_CORECLR
+                    string sCoreLibName = "System.Private.CoreLib";
```

Comment:
We should have it as C# constant somewhere to avoid these ifdefs, similar to how you have done it for the unmanaged side.

E.g. 
```
internal class CoreLib { 
#if FEATURE_CORECLR
    public const string Name = "System.Private.CoreLib";
    ... other constants as necessary ...
#else
    public const string Name = "mscorlib";
    ... other constants as necessary ...
#endif
}
```